frontend/ui/widget/inputtext.lua

```diff
@@ -641,8 +657,12 @@ function InputText:onKeyPress(key)
     end
     if not handled and (key["ScreenKB"] or key["Shift"]) then
         handled = true
-        if key["Back"] then
+        if key["Back"] and Device:hasScreenKB() then
             self:delChar()
+        elseif key["Back"] and Device:hasSymKey() then
+            self:delToStartOfLine()
+        elseif key["Del"] and Device:hasSymKey() then
+            self:delWord()
         elseif key["Left"] then
             self:leftChar()
```

Comment:
This branch was added previously - and I forgot what this was about, wondering why you would not handle Shift+Back when not hasScreenKB and not hasSymKey (ie. the emulator I guess)? But Shift+Back seems to work, not sure why.
Anyway, this branch might need some comment explaining when its use case.

Reply:
It was about having control over the cursor when a virtual keyboard is on screen. As dpad would then be used to move through the virtual keys, mod+dpad et al would let you actually interact with cursor.

> But Shift+Back seems to work, not sure why

what do you mean?

> why you would not handle Shift+Back when not hasScreenKB and not hasSymKey

there is another case for sdl doing the same already, ctrl+something

Reply:
> > But Shift+Back seems to work, not sure why
> 
> what do you mean?

I meant that Shift+Back works on my emulator.
Looking at this bit of code, I think (wrongly may be) that when I Shift+Back, I must enter that branch on line 658. But in it, not having hasScreenKB nor hasSymKey , Back shouldn't be handled - so I'm surprise it works.
But may be it's because it's handled in another branch, haven't looked.

Reply:
are you emulating a kindle? what does it do shift+back exactly?

Reply:
No, just a regular emulator. Shift+Back just do as Back, it deletes the previous char.
Well, actually no:
- Keyboard-Shift + VKB-Back does delete the previous char (but just as Keyboard-Shift + lowercase `t` outputs a lowercase `t` :) Keyboard-Shift has no effect.
- Keyboard-Shift + Keyboard-Back does nothing (and keeps doing nothing if I add a elseif in that snippet.

Well, not interested in digging into that, nevermind.